src/scripts/components/NoticesLoop.js

```diff
@@ -4,10 +4,15 @@ import { Notice } from './Notice';
  * Returns a list of notices
  * each notice is a component that has a key, an id, an image, an additional class name, and an onDismiss function
  *
- * @param {Object} prop
- * @param {Array}  prop.notices - An array of objects that contain the notice data.
+ * @param {Object}                      prop
+ * @param {import('../store').Notice[]} prop.notices - An array of objects that contain the notice data.
```

Comment:
It would be preferable to import this at the top of the file. It provides a cleaner looking function signature

```suggestion
 * @param {Notice[]} prop.notices - An array of objects that contain the notice data.
```

like:

```js
/**
 * @typedef {import('../store').Notice} Notice
 */
```
```

Reply:
I realize I'm the one that wrote the original, and now I remember why it is like that. There is a name conflict between `Notice` as a type and as a react component if the type is imported like a suggested. Leaving it as is would be best.

Reply:
Since there is a [`Notice` component](https://wordpress.github.io/gutenberg/?path=/docs/components-notice--default) in Gutenberg already- is the plan to rename this or to eventually use the core `Notice`?

Reply:
Not for this component. Though this could be renamed to be more specific. I believe the Gutenberg `Notice` is a toast notification. @erikyo and I have talked about supporting toast notifications, and I see real benefits from using the design system and components of Gutenberg.

Reply:
Agree! Also we would have to heavily modify the notifice component to make it look like how we need it and I don't know if it's worth it... if I'm not mistaken [this one](https://wordpress.github.io/gutenberg/?path=/docs/components-notice--default) does not foresee the use of an image or a title for example, and even aesthetically similar we can hardly use it!

Reply:
Yeah I agree that the core component would have to be heavily updated to work with this project- images, action items, etc- but at some point I could imagine that we wouldn't want two components with the same name.

Core does have a "[Snackbar](https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--default)" which is essentially the toast component you mentioned. I agree, that could be a benefit to incorporate at some point.

Reply:
yes indeed! with @johnhooks we identified also another additional "notification type", the **modal window** that isn't covered by the original mockup of wp-notification too (but somehow present in Gutenberg components).

Reply:
@bacoords ah you are right, I was thinking of the Snackbar. And I agree, we should work to expand `@wordpress/components` and contribute back to reduce duplication and keep building out the component library. Though I think we have a lot of work to do and need to figure out what works first.